Content.Server/Administration/PlaytimeManager.cs

```diff
@@ -0,0 +1,180 @@
+    public void Shutdown()
+    {
+        var initialUpdateInterval = UpdateInterval;
+
+        // Give it one last update and then pause until the next round begins.
+        Ticking = false;
+        //This Max() should never be necessary but better safe than sorry.
+        UpdateInterval = Math.Max(UpdateInterval -= SecondsSinceUpdate, 0);
+        UpdatePlaytimes();
+
+        // Reset everything for next round.
+        UpdateInterval = initialUpdateInterval;
+        SecondsSinceUpdate = 0;
+    }
+
+    /// <summary>
+    ///     Returns the living and total playtime for a <see cref="IPlayerSession"/> in minutes.
+    /// </summary>
+    public void FetchPlaytime(IPlayerSession session, out int livingTime, out int totalTime)
+    {
+        var data = session.ContentData();
+        if (data is null)
+        {
+            livingTime = 0;
+            totalTime = 0;
+            return;
+        }
+        livingTime = data.LivingPlaytime;
+        totalTime = data.TotalPlaytime;
+    }
+
+    /// <summary>
+    ///     Handles actual playtime updates in the <see cref="PlayerData"/> and database.
+    /// </summary>
+    private void UpdatePlaytimes()
+    {
+        var intervalSpan = TimeSpan.FromSeconds(UpdateInterval);
+
+        foreach (IPlayerSession session in _playerManager.GetAllPlayers())
+        {
+            var data = session.ContentData();
+            if (data is null) continue;
+
+            switch (session.Status)
+            {
+                // Don't let them lose time between last update and disconnect
+                case SessionStatus.Disconnected:
+                {
+                    var difference = _gameTiming.RealTime - data.DisconnectTime;
+                    // Checks to make sure we only update playtime once after DC
+                    if (difference is not null && difference.Value < intervalSpan)
+                    {
+                        AddTime(session, difference.Value);
+                    }
+                    else continue;
+                    break;
+                }
+                case SessionStatus.Connected:
+                case SessionStatus.InGame:
+                {
+                    //Disconnected sessions can be considered AFK so we check this here.
+                    if (_afkManager.IsAfk(session)) continue;
+
+                    var difference = _gameTiming.RealTime - data.JoinTime;
+                    //They joined since the last update and only get partial time
+                    if (difference < intervalSpan)
+                    {
+                        AddTime(session, difference);
+                    }
+                    //They get the full interval
+                    else
+                    {
+                        AddTime(session, intervalSpan);
+                    }
+
+                    break;
+                }
+
+            }
+
+            _dbManager.SavePlaytimeAsync(session.UserId, data.TotalPlaytime, data.LivingPlaytime);
```

Comment:
1. You are discarding the async `Task` here. Do not do that, make the method `async void` instead and `await` it.
2. This will cause a separate DB connection and transaction for each individual player. Group these so you can send all updates to the DB in one batch.

Reply:
You should check with `ServerPreferencesManager.ShouldStorePrefs` to check whether to even bother saving data for a particular player.